MaterialLib/SolidModels/Ehlers-impl.h

```diff
@@ -560,6 +547,107 @@ typename SolidEhlers<DisplacementDim>::KelvinVector predict_sigma(
     return sigma_D - pressure * Invariants::identity2;
 }
 
+template <typename ResidualVector, typename KelvinVector>
+std::tuple<KelvinVector, PlasticStrain<KelvinVector>, double>
+splitSolutionVector(ResidualVector const& solution)
```

Comment:
Is that documented somewhere? In particular the assumed layout of the `solution` would be interesting.